### ISSUE TITLE
@LOCALESTATEDIR@ incorrectly substituted in generated files

### ISSUE THREAD

**Issue by mrpapercut** [DinoTools/dionaea#201]
##### ISSUE TYPE
 - Bug Report
 - Documentation Report

##### DIONAEA VERSION
```
Dionaea Version 0.7.0
Compiled on Linux/x86_64 at May 13 2018 10:49:03 with gcc 5.4.0 20160609
Started on computer running Linux/x86_64 release 4.4.0-124-generic
```

##### CONFIGURATION
Default configuration after following installation instructions as described in docs, for Ubuntu 16.04

##### OS / ENVIRONMENT
 - Ubuntu 16.04.4 LTS

##### SUMMARY
Installing Dionaea on a clean Ubuntu 16.04 machine using the described instructions. The building succeeds without errors or warnings. When running `/opt/dionaea/bin/dionaea` I get a lot of errors for missing/unreachable files and/or permission issues. Looking at dionaea.cfg, ihandlers-available/\*.yaml and services-available/\*.yaml it appears folderpaths are incorrect.

Take file `http.yaml` as example. In the original config file `./conf/services/http.yaml.in` the config->root path is:
`"@LOCALESTATEDIR@/dionaea/roots/www"`
But the generated file `/etc/opt/dionaea/dionaea/services-available/http.yaml` lists config->root path as:
`"/dionaea/roots/www"` where you would expect it to be `""/var/opt/dionaea/lib/dionaea/roots/www"`

In addition to incorrect paths in *.yaml files, the dionaea.cfg variable "download.dir" is also incorrect.
And the folders in `/var/opt/dionaea/lib/dionaea/` are also not created. I expected the following folders here:
- binaries
- bistreams
- roots
    - ftp
    - tftp
    - upnp
    - www
- share

I have tried to run the `cmake` command with the additional parameter `-DCMAKE_INSTALL_FULL_LOCALSTATEDIR:PATH=/var/opt/dionaea/lib` but this didn't work either (admittedly, I'm not very familiar with cmake)

##### STEPS TO REPRODUCE
- Clean Ubuntu16.04 install
- Follow install instructions:
  - `sudo apt-get install git && git clone https://github.com/DinoTools/dionaea.git`
  - `cd dionaea && mkdir build && cd build`
  - `sudo apt-get install build-essential cmake check cython3 libcurl4-openssl-dev libemu-dev libev-dev libglib2.0-dev libloudmouth1-dev libnetfilter-queue-dev libnl-3-dev libpcap-dev libssl-dev libtool libudns-dev python3 python3-dev python3-bson python3-yaml ttf-liberation`
  - `make`
  - `sudo make install`
- Run dionaea with `sudo /opt/dionaea/bin/dionaea`

##### EXPECTED RESULTS
Dionaea running

##### ACTUAL RESULTS
Because `@LOCALESTATEDIR@` is incorrectly substituted, all paths where this variable is used are now incorrect, leading to a lot of warnings & errors:
```
dev@ubuntu:/opt/dionaea$ sudo bin/dionaea

Dionaea Version 0.7.0
Compiled on Linux/x86_64 at May 13 2018 12:14:45 with gcc 5.4.0 20160609
Started on ubuntu running Linux/x86_64 release 4.4.0-124-generic

[13052018 12:25:08] pptp dionaea/pptp/pptp.py:73: No config provided. Using default values
[13052018 12:25:08] http dionaea/http.py:421: Root path '/dionaea/roots/www' is not a directory
[13052018 12:25:08] http dionaea/http.py:421: Root path '/dionaea/roots/www' is not a directory
[13052018 12:25:08] ftp dionaea/ftp.py:155: The basedir '/dionaea/roots/ftp' is not a directory
[13052018 12:25:08] tftp dionaea/tftp.py:75: Root path '/dionaea/roots/tftp' is not a directory
[13052018 12:25:08] services dionaea/services.py:68: Unable to start service
Traceback (most recent call last):
  File "/opt/dionaea/lib/dionaea/python/dionaea/services.py", line 66, in start
    daemons = service.start(addr, iface=iface, config=srv.get("config", {}))
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/__init__.py", line 89, in start
    daemon = SipSession(proto=proto, config=config)
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/__init__.py", line 574, in __init__
    self.config = SipConfig(config=config)
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/extras.py", line 82, in __init__
    self._conn = sqlite3.connect(self.users)
sqlite3.OperationalError: unable to open database file
[13052018 12:25:08] pptp dionaea/pptp/pptp.py:73: No config provided. Using default values
[13052018 12:25:08] http dionaea/http.py:421: Root path '/dionaea/roots/www' is not a directory
[13052018 12:25:08] http dionaea/http.py:421: Root path '/dionaea/roots/www' is not a directory
[13052018 12:25:08] ftp dionaea/ftp.py:155: The basedir '/dionaea/roots/ftp' is not a directory
[13052018 12:25:08] tftp dionaea/tftp.py:75: Root path '/dionaea/roots/tftp' is not a directory
[13052018 12:25:08] services dionaea/services.py:68: Unable to start service
Traceback (most recent call last):
  File "/opt/dionaea/lib/dionaea/python/dionaea/services.py", line 66, in start
    daemons = service.start(addr, iface=iface, config=srv.get("config", {}))
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/__init__.py", line 89, in start
    daemon = SipSession(proto=proto, config=config)
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/__init__.py", line 574, in __init__
    self.config = SipConfig(config=config)
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/extras.py", line 82, in __init__
    self._conn = sqlite3.connect(self.users)
sqlite3.OperationalError: unable to open database file
[13052018 12:25:08] pptp dionaea/pptp/pptp.py:73: No config provided. Using default values
[13052018 12:25:08] http dionaea/http.py:421: Root path '/dionaea/roots/www' is not a directory
[13052018 12:25:08] http dionaea/http.py:421: Root path '/dionaea/roots/www' is not a directory
[13052018 12:25:08] ftp dionaea/ftp.py:155: The basedir '/dionaea/roots/ftp' is not a directory
[13052018 12:25:08] tftp dionaea/tftp.py:75: Root path '/dionaea/roots/tftp' is not a directory
[13052018 12:25:08] services dionaea/services.py:68: Unable to start service
Traceback (most recent call last):
  File "/opt/dionaea/lib/dionaea/python/dionaea/services.py", line 66, in start
    daemons = service.start(addr, iface=iface, config=srv.get("config", {}))
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/__init__.py", line 89, in start
    daemon = SipSession(proto=proto, config=config)
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/__init__.py", line 574, in __init__
    self.config = SipConfig(config=config)
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/extras.py", line 82, in __init__
    self._conn = sqlite3.connect(self.users)
sqlite3.OperationalError: unable to open database file
[13052018 12:25:08] pptp dionaea/pptp/pptp.py:73: No config provided. Using default values
[13052018 12:25:08] http dionaea/http.py:421: Root path '/dionaea/roots/www' is not a directory
[13052018 12:25:08] http dionaea/http.py:421: Root path '/dionaea/roots/www' is not a directory
[13052018 12:25:08] ftp dionaea/ftp.py:155: The basedir '/dionaea/roots/ftp' is not a directory
[13052018 12:25:08] tftp dionaea/tftp.py:75: Root path '/dionaea/roots/tftp' is not a directory
[13052018 12:25:08] services dionaea/services.py:68: Unable to start service
Traceback (most recent call last):
  File "/opt/dionaea/lib/dionaea/python/dionaea/services.py", line 66, in start
    daemons = service.start(addr, iface=iface, config=srv.get("config", {}))
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/__init__.py", line 89, in start
    daemon = SipSession(proto=proto, config=config)
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/__init__.py", line 574, in __init__
    self.config = SipConfig(config=config)
  File "/opt/dionaea/lib/dionaea/python/dionaea/sip/extras.py", line 82, in __init__
    self._conn = sqlite3.connect(self.users)
sqlite3.OperationalError: unable to open database file
[13052018 12:25:08] store dionaea/store.py:48: '/lib/dionaea/binaries/' is not a directory
[13052018 12:25:08] python /home/dev/dionaea/modules/python/module.c:1035: OperationalError at unable to open database file
[13052018 12:25:08] python /home/dev/dionaea/modules/python/module.c:1060: /opt/dionaea/lib/dionaea/python/dionaea/logsql.py:62 in None
[13052018 12:25:08] python /home/dev/dionaea/modules/python/module.c:1061:       start
[13052018 12:25:08] python /home/dev/dionaea/modules/python/module.c:1060: /opt/dionaea/lib/dionaea/python/dionaea/ihandlers.py:77 in None
[13052018 12:25:08] python /home/dev/dionaea/modules/python/module.c:1061:       start
```

#### CURRENT WORKAROUND
In order to get Dionaea running, I need to manually create the folders in `/var/opt/dionaea/lib/dionaea/`, and change all paths in services-available/\*.yaml & ihandlers-available/\*.yaml

#### SUGGESTED SOLUTION
If `@LOCALESTATEDIR@` is correctly substituted during the build, it will probably work as expected

**Comment by phibos** [comment]
Thanks for reporting the issues and for testing the cmake build scripts. The migration from autoconf to cmake is still in progress. Please feel free to have a look at the PR #202 for additional fixes and improvements for cmake.

At the moment it is still possible to use autoconf. For more information feel free to have a look at the documentation. https://dionaea.readthedocs.io/en/0.7.0/installation.html

**Comment by phibos** [comment]
This should have been fixed a few seconds ago with PR #202.
Feel free to reopen or create a new issue if there are still errors
